core/src/main/scala/ai/h2o/sparkling/backend/utils/H2OClientUtils.scala

```diff
@@ -140,4 +141,23 @@ object H2OClientUtils extends SharedBackendUtils {
     }
     None
   }
+
+  /**
+   * This method should be used whenever the Frame needs to be updated. This method ensures to use proper
+   * locking mechanism.
+   *
+   * @param fr frame to update
+   * @param f  function to run on the frame
+   * @tparam T H2O Frame Type
+   * @return returns the updated frame
+   */
+  def withLockAndUpdate[T <: Frame](fr: T)(f: T => Any): T = {
```

Comment:
Keep the few specific client related methods in this object so they are not scattered around the code